src/io/path_manager.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <cstdint>
+#include <filesystem>
+
+namespace openeaw::io {
+
+enum InstallationTypes : std::uint8_t
+{
+    steam = 1,
+};
+
+/**
+ * @brief this class handles path management and finding of installed versions and mods.
+ */
+class PathManager
+{
+private:
+    /* data */
+
+public:
+    static std::filesystem::path get_install_path(InstallationTypes type);
```

Comment:
I'm not sure if we should let the caller decide. Shouldn't this class try to find the installation path, whatever might be installed on the system? Sure, initially it can just check Steam, but eventually we can support more and should have a function like `get_install_path()`, which just returns the path it finds.

Reply:
Can do, perhaps with an option to enable/disable certain checks. Basically, we should try to minimize the code required by the callers (main game, tooling). But let's leave that as an enhancement for later. For now, it just picks the 'best' path itself (i.e. the first it can find).